target/verilator/verilator.mk

```diff
@@ -0,0 +1,77 @@
+# Copyright 2022 ETH Zurich and University of Bologna.
+# Licensed under the Apache License, Version 2.0, see LICENSE for details.
+# SPDX-License-Identifier: Apache-2.0
+
+# Jannis Schönleber, ETH Zurich
+
+VERILATOR_ROOT = $(CHS_ROOT)/target/verilator/install/verilator
+VERILATOR_SRC_PATH = $(CHS_ROOT)/target/verilator/src
+export VERILATOR_ROOT
+VERILATOR = $(VERILATOR_ROOT)/bin/verilator
+
+RISCV = /usr/pack/riscv-1.0-kgf/riscv64-gcc-11.2.0
+
+C_INCLUDE_PATH=${RISCV}/include
+CPLUS_INCLUDE_PATH=${RISCV}/include
+export C_INCLUDE_PATH
+export CPLUS_INCLUDE_PATH
+
+BENDER 		?= bender	# https://github.com/pulp-platform/bender
+MORTY			?= morty	# https://github.com/pulp-platform/morty
+
+include $(CHS_ROOT)/target/verilator/tools.mk
+
+VERILATOR_FLAGS += --cc --no-timing
+VERILATOR_FLAGS += --error-limit 4419 --unroll-count 256 -Wno-fatal
+VERILATOR_FLAGS += -Wno-WIDTHEXPAND -Wno-WIDTHTRUNC -Wno-WIDTHCONCAT -Wno-PINMISSING -Wno-MODDUP -Wno-UNOPTFLAT -Wno-BLKANDNBLK -Wno-UNUSED -Wno-ASCRANGE
+VERILATOR_EXE += --exe cheshire.cpp
+
+VERILATOR_CFILES += $(VERILATOR_SRC_PATH)/cheshire.cpp
+VERILATOR_FESVR = -LDFLAGS "-L$(RISCV)/lib -Wl,-rpath,$(RISCV)/lib -lfesvr"
+
+VERILATOR_OUTPUT = --Mdir $(CHS_ROOT)/target/verilator/obj_dir
+
+BENDER = bender
+BENDER_FLIST = bender script flist -t sim -t cv64a6_imafdc_sv39 -t test -t verilator -t cva6 -D VERILATOR=1 > flist.txt
+BENDER_SCRIPT = $(shell bender script verilator -t sim -t cv64a6_imafdc_sv39 -t verilator -t cva6 -D VERILATOR=1)
+
+# VERILATOR_INPUT = $(VERILATOR_SRC_PATH)/cheshire_testharness.sv
+VERILATOR_TOPMODULE = --top-module cheshire_testharness
+
+# Project variables
+TOP_DESIGN	= cheshire_testharness
+PICKLE_FILE := $(CHS_ROOT)/target/verilator/build/$(TOP_DESIGN).pickle.sv
+
+
+$(CHS_ROOT)/target/verilator/build/sources.json:
+	@mkdir -p $(@D)
+	$(BENDER) sources -f -t sim -t cv64a6_imafdc_sv39 -t cva6 -t verilator > $@
```

Comment:
What is the reason again that `bender script verilator` doesn't work?

Reply:
At the moment (if I remember correctly), it is out of convenience due to morty being able to remove unused stuff, and we have to patch some things, which makes it easier with morty. But the goal should be to use `bender script verilator` at some point.